### PR TITLE
[11.x] feat: introduce Number::divide()

### PR DESCRIPTION
**Pull request by sts-ryan-holton**
Hi :wave:

I'm opening another PR here to introduce a nice handy `Number::divide()` method that takes in two values and returns the result. It takes into account the "Division by zero" error which can often be encountered in projects. It's inherited from a function I wrote in my own project that I've used in a number of places.

Starting super simple here, not overcomplicating it, we could in the future introduce a `precision` flag, but wanted to keep this as lightweight as possible!

I've used this over 70 times in my application

### CODE REVIEW

**Comment by rodrigopedra** [comment]
When the denominator is zero, then it defaults the division to zero? 

What is behind this logic?

File: src/Illuminate/Support/Number.php [hunk 1]
```diff
@@ -150,6 +150,17 @@ public static function fileSize(int|float $bytes, int $precision = 0, ?int $maxP
         return sprintf('%s %s', static::format($bytes, $precision, $maxPrecision), $units[$i]);
     }
 
+    /**
+     * Divide a number.
+     *
+     * @param  int|float  $numerator
+     * @param  int  $denominator
+     */
+    public static function divide(int|float $numerator, int $denominator)
```
Comment (JesusValera) [inline src/Illuminate/Support/Number.php:159]: The denominator could also be a `float`, no? :octocat:

Comment (sts-ryan-holton) [reply src/Illuminate/Support/Number.php]: @JesusValera Good point! Thanks for the suggestion, initially didn't even think of that, I've made that change and pushed!

**Comment by taylorotwell** [comment]
Thanks for your pull request to Laravel!

Unfortunately, I'm going to delay merging this code for now. To preserve our ability to adequately maintain the framework, we need to be very careful regarding the amount of code we include. 

If applicable, please consider releasing your code as a package so that the community can still take advantage of your contributions!

**Pull request closed without merging**